src/compas/geometry/_transformations/extend.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+from compas.geometry import Translation
+from compas.geometry import Line
+from compas.geometry import Polyline
+from compas.geometry import Vector
+from compas.geometry import Point
+
+__all__ = [
+    'extend_line',
+    'extend_polyline',
+]
+
+
+def extend_line(line, start_extension=0, end_extension=0):
+    """Extend the given line from one end or the other, or both, depending on the given values
+
+    Parameters
+    ----------
+    line : tuple
+        Two points defining the line.
+    start_extension : float
+        The extension distance at the start of the line as float.
+    end_extension : float
+        The extension distance at the end of the line as float.
+
+    Returns
+    -------
+    extended line : tuple
+        Two points defining the offset line.
+
+    Examples
+    --------
+    >>> line = Line([0.0,0.0,0.0],[1.0,0.0,0.0])
+    >>> extended_line = extend_line(line, 1, 1)
+    Line([-1.0,0.0,0.0],[2.0,0.0,0.0])
```

Comment:
Same here with the missing whitespace
```suggestion
    Line([-1.0, 0.0, 0.0], [2.0, 0.0, 0.0])
```